app/src/main/java/com/github/libretube/ui/preferences/BackupRestoreSettings.kt

```diff
@@ -77,18 +87,54 @@ class BackupRestoreSettings : BasePreferenceFragment() {
         }
     }
 
+    private fun createDialog(
+        title: String,
+        items: List<String>,
+        onConfirm: (Int) -> Unit
+    ) {
+        var selectedIndex = 0
+        MaterialAlertDialogBuilder(this.requireContext())
+            .setTitle(title)
+            .setSingleChoiceItems(items.toTypedArray(), selectedIndex) { _, i ->
+                selectedIndex = i
+            }
+            .setPositiveButton(
+                R.string.okay
+            ) { _, _ -> onConfirm(selectedIndex) }
+            .setNegativeButton(R.string.cancel) { _, _ -> }
+            .show()
+    }
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.import_export_settings, rootKey)
 
         val importSubscriptions = findPreference<Preference>("import_subscriptions")
         importSubscriptions?.setOnPreferenceClickListener {
-            getSubscriptionsFile.launch("*/*")
+            val importOptions = SupportedClient.values().toList()
+                .map { it.toString().lowercase().replaceFirstChar(Char::titlecase) }
+            createDialog("Import subscriptions from", importOptions) {
+                try {
+                    selectedClient = SupportedClient.fromInt(it)
+                } catch (e: NoSuchElementException) {
+                    Log.e(TAG(), e.toString())
+                }
+                getSubscriptionsFile.launch("*/*")
+            }
             true
         }
 
         val exportSubscriptions = findPreference<Preference>("export_subscriptions")
         exportSubscriptions?.setOnPreferenceClickListener {
-            createSubscriptionsFile.launch("subscriptions.json")
+            val exportOptions = SupportedClient.values().toList()
+                .map { it.toString().lowercase().replaceFirstChar(Char::titlecase) }
+            createDialog("Export subscriptions to", exportOptions) {
+                try {
```

Comment:
(same as above)
That try catch isn't needed since we only allow the user to pick available options.